jwst/resample/resample.py

```diff
@@ -1,357 +1,592 @@
 
-        if pscale is None:
-            pscale = np.rad2deg(np.sqrt(self.output_pix_area))
-            log.info(f'Computed output pixel scale: {3600.0 * pscale} arcsec.')
+            "level": model.meta.background.level,  # sky level
+            "subtracted": model.meta.background.subtracted,
 
-        self.pscale = pscale  # in deg
+            # spectroscopy-specific:
+            "instrument_name": model.meta.instrument.name,
+            "exposure_type": model.meta.exposure.type,
+        }
 
-        log.debug(f"Output mosaic size: {tuple(self.output_wcs.pixel_shape)}")
+        if self._enable_var:
+            model_dict["var_flat"] = model.var_flat
+            model_dict["var_rnoise"] = model.var_rnoise
+            model_dict["var_poisson"] = model.var_poisson
 
+        elif (self.weight_type is not None and
+                self.weight_type.startswith('ivm')):
+            model_dict["var_rnoise"] = model.var_rnoise
 
-    def _create_output_model(self, ref_input_model=None):
+        if self._compute_err == "driz_err":
+            model_dict["err"] = model.err
+
+        return model_dict
+
+    def create_output_jwst_model(self, ref_input_model=None):
         """ Create a new blank model and update it's meta with info from ``ref_input_model``. """
         output_model = datamodels.ImageModel(None)  # tuple(self.output_wcs.array_shape))
 
         # update meta data and wcs
         if ref_input_model is not None:
             output_model.update(ref_input_model)
         output_model.meta.wcs = self.output_wcs
-        if self.output_pix_area is not None:
-            output_model.meta.photometry.pixelarea_steradians = self.output_pix_area
-            output_model.meta.photometry.pixelarea_arcsecsq = (
-                self.output_pix_area * np.rad2deg(3600)**2
-            )
         return output_model
 
-    def do_drizzle(self, input_models):
-        """Pick the correct drizzling mode based on self.single
-        """
-        if self.single:
-            return self.resample_many_to_many(input_models)
-        else:
-            return self.resample_many_to_one(input_models)
+    def update_output_model(self, model, info_dict):
+        model.data = info_dict["data"]
+        model.wht = info_dict["wht"]
+        if self._enable_ctx:
+            model.con = info_dict["con"]
+        if self._compute_err:
+            model.err = info_dict["err"]
+        if self._enable_var:
+            model.var_rnoise = info_dict["var_rnoise"]
+            model.var_flat = info_dict["var_flat"]
+            model.var_poisson = info_dict["var_poisson"]
+
+        model.meta.wcs = info_dict["wcs"]
+        model.meta.photometry.pixelarea_steradians = info_dict["pixelarea_steradians"]
+        model.meta.photometry.pixelarea_arcsecsq = info_dict["pixelarea_arcsecsq"]
+
+        model.meta.resample.pointings = info_dict["pointings"]
+        # model.meta.resample.n_coadds = info_dict["n_coadds"]
```

Comment:
This would be helpful if we ever wanted to be able to keep adding new data models to an existing resampled image (in a different run of the step) instead of resampling all images again. It does not need to be done right now but it would not hurt to have this additional information.

Reply:
But this needs a schema update => this can be done at a later time.